src/pyFAI/io/diffmap_config.py

```diff
@@ -0,0 +1,347 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+"""Module with the configuration dataclass for diffraction mapping experiments"""
+
+__author__ = "Jérôme Kieffer"
+__contact__ = "Jerome.Kieffer@ESRF.eu"
+__license__ = "MIT"
+__copyright__ = "European Synchrotron Radiation Facility, Grenoble, France"
+__date__ = "04/04/2025"
+__status__ = "development"
+__docformat__ = 'restructuredtext'
+
+import os
+import time
+import copy
+import json
+import logging
+logger = logging.getLogger(__name__)
+from collections import namedtuple
+from .tree import TreeItem
+from .integration_config import dataclass, ClassVar, WorkerConfig, fields, asdict
+from .nexus import is_hdf5
+
+#constants:
+CURRENT_VERSION = 1  # former version were unassigned
+
+
+@dataclass
+class MotorRange:
+    """ This object represents a motor range
+
+    :param start: Begining of the movement
+    :param stop: End of the movement, included
+    :param step: Number of points (i.e. numberof steps + 1)
+    :param name: Name of the motor
+    """
+    start: float = 0.0
+    stop: float = 1.0
+    points: int = 0
+    name: str = ""
+
+    @classmethod
+    def _fromdict(cls, dico):
+        "Mirror of _asdict: take the dict and populate the tuple to be returned"
+        try:
+            obj = cls(**dico)
+        except TypeError as err:
+            logger.warning("TypeError: %s", err)
+            obj = cls(**{key: dico[key] for key in [i for i in cls._fields if i in dico]})
+        return obj
+
+    def as_dict(self):
+        """Like asdict, but possibly with some more features
+        """
+        return dict(asdict(self))
+
+    def __repr__(self):
+        return f"{self.name}: MotorRange({self.start:.3f}, {self.stop:.3f}, {self.points})"
+    @property
+    def step_size(self):
+        if self.points < 1:
+            return
+        return (self.stop-self.start)/self.points
+
+
+DataSetNT = namedtuple("DataSet", ("path", "h5", "nframes", "shape"), defaults=[None])
+
+@dataclass
+class DataSet:
+    path: str
+    h5: str = None
+    nframes: int = None
+    shape: tuple = None
+
+    def __repr__(self):
+        return f"Dataset('{self.path}', '{self.h5}', {self.nframes}, {self.shape})"
+
+    def as_tuple(self):
+        return DataSetNT(self.path, self.h5, self.nframes, self.shape)
+
+    @classmethod
+    def from_tuple(cls, tpl):
+        return cls(*tpl)
+
+    def is_hdf5(self):
+        """Return True if the object is hdf5"""
+        if self.h5 is None:
+            self.h5 = is_hdf5(self.path)
+        return bool(self.h5)
```

Comment:
Is there really a reason to cache `self.h5` ? Why not making it a computed `property` ?

Reply:
Good remark ... since I just "dataclassed" an existing class
https://github.com/silx-kit/pyFAI/blob/55d53cf195d9600689e3a29f0626b3d1b6bb917b/src/pyFAI/gui/utils/tree.py#L147
I did not re-think the way it was working.

Actually this object is strange since `self.h5` is now the path of the hdf5 file and contains no reference to the internal path. I opened an issue to think specifically about this issue: #2503